cns/restserver/restserver.go

```diff
@@ -1252,7 +1263,31 @@ func (service *HTTPRestService) getNetworkContainerResponse(req cns.GetNetworkCo
 		}
 
 		logger.Printf("pod info %+v", podInfo)
-		containerID = service.state.ContainerIDByOrchestratorContext[podInfo.PodName+podInfo.PodNamespace]
+
+		var (
+			context      = podInfo.PodName + podInfo.PodNamespace
+			dncEP        = service.GetOption(acn.OptPrivateEndpoint).(string)
+			infraVnet    = service.GetOption(acn.OptInfrastructureNetwork).(string)
+			nodeID       = service.GetOption(acn.OptNodeID).(string)
+			isManagedDnc = dncEP != "" && infraVnet != "" && nodeID != ""
+		)
+
+		containerID, exists = service.state.ContainerIDByOrchestratorContext[context]
+		if !exists {
+			if isManagedDnc {
+				service.RUnlock()
+				getNetworkContainerResponse.Response.ReturnCode, getNetworkContainerResponse.Response.Message = service.SyncNodeStatus(dncEP, infraVnet, nodeID, req.OrchestratorContext)
```

Comment:
what happens if we crash in line 1279. in existing defer you would try unlock the lock which you already unlocked on 1278

Reply:
if it crashes then the unlock will be a runtime err